dvc/commands/studio.py

```diff
@@ -0,0 +1,172 @@
+import argparse
+import os
+
+from funcy import get_in
+
+from dvc.cli.utils import append_doc_link, fix_subparsers
+from dvc.commands.config import CmdConfig
+from dvc.log import logger
+
+logger = logger.getChild(__name__)
+
+DEFAULT_SCOPES = "live,dvc_experiment,view_url,dql,download_model"
+
+
+class CmdStudioLogin(CmdConfig):
+    def run(self):
+        from dvc_studio_client.auth import StudioAuthError, initiate_authorization
+
+        from dvc.env import DVC_STUDIO_URL
+        from dvc.ui import ui
+        from dvc.utils.studio import STUDIO_URL
+
+        name = self.args.name
+        hostname = self.args.hostname or os.environ.get(DVC_STUDIO_URL) or STUDIO_URL
+        scopes = self.args.scopes or DEFAULT_SCOPES
+
+        try:
+            token_name, access_token = initiate_authorization(
```

Comment:
On studio-client side, let's rename this to a more meaningful name. `get_access_token()` perhaps? Or if you have something better.